scripts/src/indexfile/index.py

```diff
@@ -38,5 +43,97 @@ def get_chart_info(tar_name):
+    for entry, charts in index_dct["entries"].items():
+        for chart in charts:
+            chart_info = {}
+            chart_info["name"] = chart['name']
+            chart_info["version"] = chart["version"]
+            chart_info["providerType"] = chart["annotations"]["charts.openshift.io/providerType"]
+            chart_info["provider"] =  entry.removesuffix(f'-{chart["name"]}')
+            #print(f'[INFO] found chart : {chart_info["provider"]} {chart["name"]} {chart["version"]} ')
+            if 'charts.openshift.io/supportedOpenShiftVersions' in chart["annotations"]:
+                chart_info["supportedOCP"] = chart["annotations"]["charts.openshift.io/supportedOpenShiftVersions"]
+            else:
+                chart_info["supportedOCP"] = ""
+            if "kubeVersion" in chart:
+                chart_info["kubeVersion"] = chart["kubeVersion"]
+            else:
+                chart_info["kubeVersion"] =""
+            chart_info_list.append(chart_info)
+
+    return chart_info_list
+
+def get_latest_charts():
+    chart_list = get_charts_info()
+
+    print(f"{len(chart_list)} charts found in Index file")
+
+    chart_in_process = {"name" : ""}
+    chart_latest_version = ""
+    latest_charts = []
+
+    for index,chart in enumerate(chart_list):
+        chart_name = chart["name"]
+        #print(f'[INFO] look for latest chart : {chart_name} {chart["version"]}')
+        if chart_name == chart_in_process["name"]:
+            new_version = semantic_version.Version.coerce(chart["version"])
+            #print(f'   [INFO] compare chart versions : {new_version}({chart["version"]}) : {chart_latest_version}')
+            if new_version > chart_latest_version:
+                #print(f'   [INFO] a new latest chart version : {new_version}')
+                chart_latest_version = new_version
+                chart_in_process = chart
+        else:
+            if chart_in_process["name"] != "":
+                #print(f'   [INFO] chart completed : {chart_in_process["name"]} {chart_in_process["version"]}')
+                latest_charts.append(chart_in_process)
+
+                #print(f'[INFO] new  chart found : {chart_name} {chart["version"]}')
+                chart_in_process = chart
+                chart_version = chart["version"]
+                if chart_version.startswith("v"):
+                    chart_version = chart_version[1:]
+                chart_latest_version = semantic_version.Version.coerce(chart_version)
+            else:
+                chart_in_process = chart
+
+        if index+1 == len(chart_list):
+            #print(f'   [INFO] last chart completed : {chart_in_process["name"]} {chart_in_process["version"]}')
+            latest_charts.append(chart_in_process)
+
+    return latest_charts
+
+
 if __name__ == "__main__":
-    get_chart_info("redhat-dotnet-0.0.1")
\ No newline at end of file
+    get_chart_info("redhat-dotnet-0.0.1")
```

Comment:
It would have been good to see some unit tests for the new functions added to the modules. I see it missing for other modules as well. Anyway we can scope that as part of other story.

Reply:
Yes, lets scope it in another story. Don't have enough time to rewrite and do all the extra testing at the moment.